data/lib/service/user/user_service.dart

```diff
@@ -117,6 +141,22 @@ class UserService {
     }
   }
 
+  Future<UserModel?> getUserByPhoneNumber(String number) async {
+    try {
+      final snapshot = await _userRef
+          .where(FireStoreConst.phone, isEqualTo: number)
+          .limit(1)
```

Comment:
Why are you using a limit? I think there's only one user in the database with this number.